File: customer/src/main/java/br/com/accenture/wallet/customer/controller/CustomerErrorHandler.java
package br.com.accenture.wallet.customer.controller;

import br.com.accenture.wallet.customer.service.ResourceNotFoundException;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@ControllerAdvice
public class CustomerErrorHandler {

    @ResponseBody
    @ResponseStatus(HttpStatus.NOT_FOUND)
    @ExceptionHandler(ResourceNotFoundException.class)
    public FailModel resourceNotFound(ResourceNotFoundException ex) {
        return new FailModel("fail", ex.getMessage());
    }

    class FailModel {
        public int code;
        public String type;
        public String message;

        public FailModel(String type, String msg) {
            this.type = type;
            this.message = msg;
        }

        public int getCode() {
            return code;
        }

        public String getType() {
            return type;
        }

        public String getMessage() {
            return message;
        }
    }

}

File: customer/src/main/kotlin/br/com/accenture/wallet/customer/controller/CustomerErrorHandler.kt
package br.com.accenture.wallet.customer.controller

import br.com.accenture.wallet.customer.service.ResourceNotFoundException
import org.springframework.http.HttpStatus.NOT_FOUND
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.ResponseStatus

@ControllerAdvice
class CustomerErrorHandler {

    @ResponseBody
    @ResponseStatus(NOT_FOUND)
    @ExceptionHandler(ResourceNotFoundException::class)
    fun resourceNotFound(ex: ResourceNotFoundException): FailModel {
        return FailModel("fail", ex.message)
    }

    inner class FailModel(val type: String, val message: String?) {
        var code: Int = 0
    }

}
